datapackage/resource.py

```diff
@@ -312,38 +310,21 @@ def _parse_inline_data(self):
 
         return data
 
+    def _iter_from_tabulator(self, table, schema):
+        if schema is not None:
+            schema = jsontableschema.model.SchemaModel(schema)
```

Comment:
It's a bit confusing to reuse the `schema` variable here. I would prefer setting a new variable (e.g. `schema_model`) instead, so it's clearer that the later uses of this variable (e.g. lines 317 and 318) are refering to this new `SchemaModel`, and not the `schema` received as parameter.